### PR TITLE
Updated oniguruma dependency to support node.js 6

### LINKED ISSUE #72: Update oniguruma dependency to v6.1.0 to avoid stack trace on node >= v6.0.0
`(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated`

Is solved if oniguruma is updated to v6.1.0

### PR DESCRIPTION
**Pull request by janb87**
Fixes #72

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ notifications:
     on_failure: change
 
 node_js:
+  - 6
```
Comment (50Wliu) [inline .travis.yml]: Should we always test the latest Node version, eg using `node`?

Comment (janb87) [inline .travis.yml]: Would just list the versions explicitly. Most projects do it like that.

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ notifications:
...
+  - 5
```
Comment (addaleax) [inline .travis.yml]: v4 will be supported for a much longer time (see e.g. the [LTS schedule](https://github.com/nodejs/LTS)) than v5, so if there’s a second additional version here, it’s probably better to choose v4 over v5.

**Comment by zeke** [comment]
@50Wliu I see you added the `needs-testing` label. What testing need to be done before this can be shipped?

**Comment by 50Wliu** [comment]
Just some basic regression testing should suffice, though you probably know more about what needs to be done than I do :).

**Comment by zeke** [comment]
oniguruma doesn't seem to have a changelog or release notes.

Hey @zcbenz, @alexandrudima, and @evanlucas: do you know what breaking change happened in oniguruma from 5 to 6?

https://github.com/atom/node-oniguruma/commits/master

**Comment by alexandrudima** [comment]
@zeke: https://github.com/atom/node-oniguruma/pull/46

And the relevant comment:

```
// This is the only way to get caching between `FindNextMatchSync` calls:
var str = new OnigString("hello world");
scanner.FindNextMatchSync(str, 0);
scanner.FindNextMatchSync(str, 1);
scanner.FindNextMatchSync(str, 2);
...
```

To benefit from caching between calls on the same scanner instance you need to create a OnigString up-front. I explain the reason behind that in the PR.

**Comment by ashleygwilliams** [comment]
hey ya'll- what can i do to help move this forward?

**Comment by zeke** [comment]
Just landed this in a new major version: 6.0.0

**Comment by ashleygwilliams** [comment]
zomg amazing!! thank you so much! 🎉

**Pull request merged**